diff a/plugins/thunar-wallpaper/twp-provider.c b/plugins/thunar-wallpaper/twp-provider.c
--- a/plugins/thunar-wallpaper/twp-provider.c
+++ b/plugins/thunar-wallpaper/twp-provider.c
@@ -51,6 +51,7 @@ static void   twp_action_set_wallpaper          (ThunarxMenuItem          *item,
 static gint   twp_get_active_workspace_number   (GdkScreen *screen);
 
 static gboolean    _has_gsettings = FALSE;
+static gboolean    _has_setbg_bodhi = FALSE;
 static GtkWidget   *main_window = NULL;
 
 struct _TwpProviderClass
@@ -100,6 +101,12 @@ twp_provider_init (TwpProvider *twp_provider)
       _has_gsettings = TRUE;
       g_free (program);
     }
+  program = g_find_program_in_path ("set-background-bodhi");
+  if (G_LIKELY (program != NULL))
+   {
+     _has_setbg_bodhi = TRUE;
+      g_free (program);
+   }
 }
 
 
@@ -149,7 +156,9 @@ twp_provider_get_file_menu_items (ThunarxMenuProvider *menu_provider,
               && (thunarx_file_info_has_mime_type (files->data, "image/jpeg")
                   || thunarx_file_info_has_mime_type (files->data, "image/png")
                   || thunarx_file_info_has_mime_type (files->data, "image/svg+xml")
-                  || thunarx_file_info_has_mime_type (files->data, "image/svg+xml-compressed")))
+                  || thunarx_file_info_has_mime_type (files->data, "image/svg+xml-compressed")
+                  || thunarx_file_info_has_mime_type (files->data, "image/jxl")
+                  || thunarx_file_info_has_mime_type (files->data, "image/avif")))
             {
               item = thunarx_menu_item_new ("Twp::setwallpaper", _("Set as wallpaper"), NULL, "preferences-desktop-wallpaper");
               g_signal_connect (item, "activate", G_CALLBACK (twp_action_set_wallpaper), files->data);
@@ -310,11 +319,22 @@ twp_action_set_wallpaper (ThunarxMenuItem *item,
           g_warning ("Failed to set wallpaper: Missing executable 'gsettings'");
         }
     }
+  else if (g_strcmp0 (desktop_type, "Enlightenment") == 0)
+   {
+     if (_has_setbg_bodhi)
+      {
+            g_debug ("set on enlightenment");
+            command = g_strdup_printf ("set-background-bodhi "
+                                                        "'%s'", file_name);
+            g_spawn_command_line_async (command, NULL);
+            g_free (command);
+     }
+   }
   else
     {
       g_warning (("Failed to set wallpaper: $XDG_CURRENT_DESKTOP Desktop type '%s' not supported by thunar wallpaper plugin."), desktop_type);
     }
-
+ 
   g_free (escaped_file_name);
   g_free (file_name);
   g_free (file_uri);
